scanners/flutter/flutterproject/pubspec_lock.go

```diff
@@ -0,0 +1,72 @@
+package flutterproject
+
+import (
+	"io"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+const pubspecLockRelPath = "pubspec.lock"
+
+type PubspecLockVersionReader struct {
+	fileOpener FileOpener
+}
+
+func NewPubspecLockVersionReader(fileOpener FileOpener) PubspecLockVersionReader {
+	return PubspecLockVersionReader{
+		fileOpener: fileOpener,
+	}
+}
+
+func (r PubspecLockVersionReader) ReadSDKVersions(projectRootDir string) (*VersionConstraint, *VersionConstraint, error) {
+	pubspecLockPth := filepath.Join(projectRootDir, pubspecLockRelPath)
+	f, err := r.fileOpener.OpenFile(pubspecLockPth)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if f == nil {
+		return nil, nil, nil
+	}
+
+	flutterVersionStr, dartVersionStr, err := parsePubspecLockSDKVersions(f)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var flutterVersion *VersionConstraint
+	if flutterVersionStr != "" {
+		flutterVersion, err = NewVersionConstraint(flutterVersionStr, PubspecLockVersionSource)
+		if err != nil {
+			return nil, nil, err
```

Comment:
Same as `pubspec.yaml`, is it really an error scenario if one of the SDK versions is missing?

From our demo app: https://github.com/bitrise-io/flutter-demo-app/blob/main/pubspec.lock

Reply:
Noticed you already cover this in the unit test and I missed the check on line 39 in my initial review.